src/screen/ConnectionsList/Connection/index.tsx

```diff
@@ -59,55 +74,140 @@ const Connection: React.FC<IConnectionProps> = ({ connection }) => {
         setIsConnecting(false)
       }
     }
-  }, [])
+  }, [isConnected, connection, setCurrentConnection, setCurrentDatabase])
+
+  const handleDisconnect = useCallback(async () => {
+    setCurrentConnection(undefined)
+    setCurrentDatabase(undefined)
+  }, [setCurrentConnection, setCurrentDatabase])
+
+  const handleRefreshDatabases = useCallback(async () => {
+    try {
+      setIsConnecting(true)
+
+      const databases = await loadConnectionDatabases(connection)
 
-  const handleSelectDatabase = useCallback((database: IDatabase) => {
-    setCurrentDatabase(database)
-  }, [])
+      setDatabases(databases)
+    } catch {
+      setIsConnectionFailed(true)
+    } finally {
+      setIsConnecting(false)
```

Comment:
Maybe we should change this state name as it's being used to control any "loading" state in the connection now and not the "connecting" state, maybe `connectionLoading` and `setConnectionLoading`